.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  legacy:
+    
+    name: "Python 2.6 Unit Tests"
+    runs-on: ubuntu-16.04
+
+    strategy:
+      fail-fast: false
+    env:
+      NOSEOPTS: "--verbose"
+
+    steps:
+
+    - name: Install Python 2.6
+      run: |
+        sudo add-apt-repository ppa:deadsnakes/ppa -y
```

Comment:
It is not clear to me who publishes this repo. There are multiple references in blog posts, so probably it is safe enough for unit tests, but we should avoid it if possible (unless we know the publisher and its reputation). couldn't we just create a python venv, upload it to storage and install it here?

Reply:
If going that route it might be better to try to get and publish a docker container image with py2.6 preinstalled. Creating a python 2.6 venv might or might not work in the test container, totally depends on the availability of the same set of libraries as the VM/container where the venv was created

Reply:
Checking whether we can create a venv should not be too difficult. This is running on ubuntu 16; if it is not possible to compile 2.6 from source there, we could try to create a VM in the closest distro to ubuntu 16 that we can find that uses 2.6 and create the venv there.

If this is too difficult we can do it on a later PR.

Reply:
Removed ETP

Reply:
BTW, Travis uses this venv on ubuntu 14, we can try to check whether there are any licensing that would prevent us from using it. https://storage.googleapis.com/travis-ci-language-archives/python/binaries/ubuntu/14.04/x86_64/python-2.6.tar.bz2

Reply:
Definitely. I think Travis actually does pretty much this for their python2.6 environment, and it's obviously (at least, imo) a much cleaner solution. Do we think this is a high enough priority to stall the PR until we have the venv built and uploaded? I can't image it would take _too_ long, but the current code is already working, and upgrading to a blobbed venv should be a simple matter of tweaking these couple of shell lines once the venev is uploaded.

Reply:
yes, we can do it on a subsequent pr

Reply:
We can also build python and run it
https://github.com/pyenv/pyenv does this very well, and there is no licensing about it. Also its clean and pyenv is well known and supported by the community.
(I tried it and it worked on Ubuntu 14-04, and took around 5 extra minutes to build)

Reply:
>https://github.com/pyenv/pyenv does this very well

Correct me if I'm wrong, but this doesnt build for you right? Just makes it easier to manager already installed python versions on the box?

Reply:
It builds for you
`pyenv install 2.6.7` will fetch the source for 2.6.7, patch it, and build it
`pyenv global 2.6.7` to set it as system version